### PR TITLE
Include the value of environment variable "DBT_DATABRICKS_INVOCATION_ENV" to the user agent entry.

### PR DESCRIPTION
**Pull request by KoningJasper**
### Description

Include the value of environment variable "DBT_DATABRICKS_INVOCATION_ENV" to the user agent entry.

### Checklist

- [x] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [ ] I have updated the `CHANGELOG.md` and added information about my change to the "dbt-databricks next" section.

### CODE REVIEW

File: dbt/adapters/databricks/connections.py [hunk 1]
```diff
@@ -39,6 +40,7 @@
 logger = AdapterLogger("Databricks")
 
 CATALOG_KEY_IN_SESSION_PROPERTIES = "databricks.catalog"
+CUSTOM_USER_AGENT = "__DBT_DATABRICKS_CUSTOM_USER_AGENT__"
```
Comment (lennartkats-db) [inline dbt/adapters/databricks/connections.py:43]: Are there precedents for the naming convention used here?

**Review by lennartkats-db — COMMENTED** [review]
This looks good; minor question, please review. Nice touch on making sure we're using the right naming convention.

File: dbt/adapters/databricks/connections.py [hunk 2]
```diff
@@ -291,6 +301,14 @@ def open(cls, connection: Connection) -> Connection:
         creds: DatabricksCredentials = connection.credentials
         exc: Optional[Exception] = None
 
+        dbt_databricks_version = __version__.version
+        user_agent_entry = f"dbt-databricks/{dbt_databricks_version}"
+
+        custom_user_agent = os.environ.get(CUSTOM_USER_AGENT)
+        if custom_user_agent is not None and len(custom_user_agent) > 0:
+            cls.validate_custom_user_agent(custom_user_agent)
+            user_agent_entry = f"dbt-databricks/{dbt_databricks_version}; {custom_user_agent}"
```
Comment (ueshin) [inline dbt/adapters/databricks/connections.py:310]: How about:

```py
user_agent_entry = f"{user_agent_entry}; {custom_user_agent}"
```

?

**Review by ueshin — COMMENTED** [review]
Could you also update the variable names or method names to adjust the env var name?

File: dbt/adapters/databricks/connections.py (hunk 1 above)
Comment (allisonwang-db) [reply dbt/adapters/databricks/connections.py:43]: :+1:

**Review by ueshin — APPROVED** [review]
LGTM.

**Comment by ueshin** [comment]
Thanks! merging.

**Pull request merged**